v2/contribute/decisions/thirdparty/0005-implicit-multitenancy-init.mdx

```diff
@@ -0,0 +1,51 @@
+---
+id: "0005"
+title: Implicit initialization of Multitenant recipe
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Implicit initialization of Multitenant recipe
+
+<DecisionHeader status="proposed" lastUpdate="2022-12-08" created="2022-12-08" deciders={["rishabhpoddar", "sattvikc"]} proposedBy={["sattvikc"]} />
+
+## Context and Problem Statement
+
+All the dynamic config management functions are exposed from the Multitenancy recipe.
+
+Should the Multitenancy recipe implicitly initialized by the SDK, or explicitly initialized by the user, for the user to be able to use those functions?
+
+## Considered Options
+
+* Multitenancy recipe is initialized explicitly by the user
+* Multitenancy recipe is initialized implicitly by the SDK
+
+
+## Decision Outcome
+
+Chosen option: **Multitenancy recipe is initialized implicitly by the SDK**, because
+
+On the backend:
+* Allows the usage of the config management functions in the Multitenancy recipe without having to initialize it.
+
+On the frontend:
+* When user is using dynamic config for a single tenant, the SDK requires Multitenant to be initialized to be able to fetch the dynamic config from the backend. Initializing implicitly gets the functionality to the user without any new configuration.
+
+*Note for Backend: The user can still explicitly initialize the recipe in case the recipe functions needs overriding.*
+
+*Note for Frontend: The user will need to explicitly initialize the recipe in case of a multi-tenant setup, to be able to provide the tenantId*
+
+## Pros and Cons of the Options
+
+### Multitenancy recipe is initialized explicitly by the user
+
+<ArgumentCon> Additional config for the user </ArgumentCon>
```

Comment:
?? But it's still an additional config in case they want to provide their own tenantId.. so be specific.

Reply:
Updated